scripts/zkai_native_attention_mlp_rmsnorm_input_fused_adapter_gate.py

```
@@ -0,0 +1,510 @@
+        payload["mutation_result"] = mutation_result(payload, context=context)
+        refresh_payload_commitment(payload)
+    return payload
+
+
+def validate_payload(payload: dict[str, Any], *, context: dict[str, Any] | None = None) -> None:
+    if context is None:
+        context = build_context()
+    summary = require_dict(payload.get("summary"), "summary")
+    comparisons = require_dict(payload.get("comparisons"), "comparisons")
+    if summary["fused_typed_bytes"] <= summary["compact_typed_bytes"]:
+        raise RmsnormInputFusedAdapterGateError("fused proof-size win is not supported")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)